components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoOrientationDelegate.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko
+
+import org.mozilla.geckoview.AllowOrDeny
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.OrientationController
+
+class GeckoOrientationDelegate(
+    private val geckoEngineSession: GeckoEngineSession
+) : OrientationController.OrientationDelegate {
+    override fun onOrientationLock(requestedOrientation: Int): GeckoResult<AllowOrDeny> {
+        geckoEngineSession.notifyObservers {
+            onOrientationLockRequest(requestedOrientation)
+        }
+        return GeckoResult.allow()
```

Comment:
Asked about this on https://bugzilla.mozilla.org/show_bug.cgi?id=1697647#c22
All preconditions are checked by GeckoView which seems to only needs us (integrators as Fenix) because we have the activity on which `setRequestedOrientation` needs to be called.